### PR TITLE
ch-remote: Add "create" command to ch-remote

### PR DESCRIPTION
**Pull request by rbradford**
Tested by:
1. Path: ch-remote --api-socket .. create test-config.json
2. stdin with "-" path: ch-remote --api-socket .. create - < test-config.json
2. stding with implicit "-": ch-remote --api-socket .. create < test-config.json

Fixes: #4250

Signed-off-by: Rob Bradford <robert.bradford@intel.com>

### CODE REVIEW

File: src/bin/ch-remote.rs [hunk 1]
```diff
@@ -309,6 +314,20 @@ fn send_migration_api_command(
...
 
+fn create_api_command(socket: &mut UnixStream, path: &str) -> Result<(), Error> {
+    let mut data = String::default();
+    if path == "-" {
+        std::io::stdin()
+            .read_to_string(&mut data)
+            .map_err(Error::ReadingStdin)?;
+    } else {
+        let mut f = std::fs::File::open(path).map_err(Error::ReadingFile)?;
+        f.read_to_string(&mut data).map_err(Error::ReadingFile)?;
```
Comment (sboeuf) [inline src/bin/ch-remote.rs:325]: Using `std::fs::read_to_string(path)` would save one line.

**Review by sboeuf — APPROVED** [review]
Looking good, but you should modify an existing integration test (or add a new one) to validate this is working correctly when using `ch-remote`.

File: src/bin/ch-remote.rs (hunk 1 above)
Comment (rbradford) [reply src/bin/ch-remote.rs:325]: Dang, I thought there was a function like that and I couldn't find it. As this got automerged. I'll address this in a follow up commit.

**Comment by scuzhanglei** [comment]
actually, I want to run cloud-hypervisor in a container, and make the  cloud-hypervisor as the entrypoint process to create a VM directly. anyway thanks for your works.

**Comment by rbradford** [comment]
> actually, I want to run cloud-hypervisor in a container, and make the cloud-hypervisor as the entrypoint process to create a VM directly. anyway thanks for your works.

@scuzhanglei Does this solution not allow you to do that?

**Comment by scuzhanglei** [comment]
yes，If I am right. this solution need run cloud-hypervisor firstly，and  use ch-remote in another process。

**Pull request merged**